File: data/compilers_comparation/src/examples/second_example/src/func20.c
#include "second_example.h" 
array_t* func20(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array200;
      if (pCounter > 0) {
         array200 = vars->data[--pCounter];
         array200->refC++;
         DEBUG_COPY(array200->id);
      } else {
         array200 = (array_t*)malloc(sizeof(array_t));
         array200->size = 2;
         array200->refC = 1;
         array200->id = 200;
         array200->data = (unsigned int*)malloc(array200->size*sizeof(unsigned int));
         memset(array200->data, 0, array200->size*sizeof(unsigned int));
         DEBUG_NEW(array200->id);
      }
      unsigned int loop59 = 0;
      unsigned int loopLimit59 = (rand()%loopsFactor)/3 + 1;
      for(; loop59 < loopLimit59; loop59++) {
         for (int i = 0; i < array200->size; i++) {
            array200->data[i]--;
         }
         for (int i = 0; i < array200->size; i++) {
            if (array200->data[i] == 96) { 
               return array200;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array200;
      array_t* array201 = func28(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array201->id);
      free(params0.data);
      array201->refC--;
      if(array201->refC == 0) {
         free(array201->data);
         free(array201);
         DEBUG_FREE(array201->id);
      }
      array200->refC--;
      if(array200->refC == 0) {
         free(array200->data);
         free(array200);
         DEBUG_FREE(array200->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array202 = func22(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array202->id);
      free(params0.data);
      array202->refC--;
      if(array202->refC == 0) {
         free(array202->data);
         free(array202);
         DEBUG_FREE(array202->id);
      }
   }
   array_t* array203;
   if (pCounter > 0) {
      array203 = vars->data[--pCounter];
      array203->refC++;
      DEBUG_COPY(array203->id);
   } else {
      array203 = (array_t*)malloc(sizeof(array_t));
      array203->size = 21;
      array203->refC = 1;
      array203->id = 203;
      array203->data = (unsigned int*)malloc(array203->size*sizeof(unsigned int));
      memset(array203->data, 0, array203->size*sizeof(unsigned int));
      DEBUG_NEW(array203->id);
   }
   return array203;
}

File: data/multilang_comparison/program_nim/src/func20.nim
proc func20*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var array232: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array232 = vars.data[pCounter]
         inc array232.refC
         debugCopy(array232.id)
      else:
         array232 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array232.size = 231
         array232.refC = 1
         array232.id = 232
         array232.data = cast[ptr UncheckedArray[uint32]](alloc(array232.size * sizeof(uint32)))
         zeroMem(array232.data, array232.size * sizeof(uint32))
         debugNew(array232.id)
      let loopLimit68 = (50) div (3) + 1
      for loop68 in 0 ..< loopLimit68:
         discard
         for i in 0 ..< array232.size:
            dec array232.data[i]
         for i in 0 ..< array232.size:
            if array232.data[i] == 40:
               return array232
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array232
      var array233: ptr ArrayT = func28(addr params0, getPath(), loopsFactor)
      debugReturn(array233.id)
      dealloc(params0.data)
      dec array233.refC
      if array233.refC == 0:
         dealloc(array233.data)
         dealloc(array233)
         debugFree(array233.id)
      dec array232.refC
      if array232.refC == 0:
         dealloc(array232.data)
         dealloc(array232)
         debugFree(array232.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array234: ptr ArrayT = func22(addr params0, getPath(), loopsFactor)
      debugReturn(array234.id)
      dealloc(params0.data)
      dec array234.refC
      if array234.refC == 0:
         dealloc(array234.data)
         dealloc(array234)
         debugFree(array234.id)
   var array235: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array235 = vars.data[pCounter]
      inc array235.refC
      debugCopy(array235.id)
   else:
      array235 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array235.size = 81
      array235.refC = 1
      array235.id = 235
      array235.data = cast[ptr UncheckedArray[uint32]](alloc(array235.size * sizeof(uint32)))
      zeroMem(array235.data, array235.size * sizeof(uint32))
      debugNew(array235.id)
   return array235

